gradle-plugin/src/main/java/com/squareup/hephaestus/plugin/CheckMixedSourceSet.kt

```diff
@@ -20,11 +18,12 @@ import java.io.File
  * these module using this task.
```

Comment:
Can you add that this operation is performed at configuration time ?

Reply:
It's not, it's called in a doFirst and this is fine:
```
    project.tasks
        .withType(KotlinCompile::class.java)
        .configureEach { compileTask ->
          compileTask.doFirst {
            // Disable precise java tracking if needed. Note that the doFirst() action only runs
            // if the task is not up to date. That's ideal, because if nothing needs to be
            // compiled, then we don't need to disable the flag.
            CheckMixedSourceSet(project, compileTask).disablePreciseJavaTrackingIfNeeded()

            compileTask.logger.info(
                "Hephaestus: Use precise java tracking: ${compileTask.usePreciseJavaTracking}"
            )
          }
        }
```

Reply:
I am surprised a `doFirst` block can access other tasks with CoCa.

Reply:
Oh, but it's the same task ! So yeah, it's all good.